lib/idx/remediate.ts

```diff
@@ -169,17 +137,20 @@ export async function remediate(
 
   // Reach to terminal state
   const terminal = isTerminalResponse(idxResponse);
-  const messages = getIdxMessages(idxResponse);
+  const messages = getMessagesFromResponse(idxResponse);
   if (terminal) {
-    return { terminal, messages };
+    return { idxResponse, terminal, messages };
   }
 
   const remediator = getRemediator(neededToProceed, values, options);
-  if (!remediator && flow === 'default') {
-    return { idxResponse };
-  }
-  
   if (!remediator) {
+    if (options.step) {
+      idxResponse = await idxResponse.proceed(options.step, values);
+      return { idxResponse };
```

Comment:
return directly instead of keep remediating? what's the scenario to handle here?

Reply:
If there is no remediator, but a step has been specified then it will proceed using the values provided. (This is the main mode the widget will use, it will provide all the values and specify the step (remediation) that it wants to resolve, so it will not error if we do not have a Remediator defined).